src/template/render.rs

```diff
@@ -264,13 +267,34 @@ impl<'a> TemplateSource<'a> for ChainTemplateSource<'a> {
             // We intentionally throw the content detection error away here,
             // because it isn't that intuitive for users and is hard to plumb
             let (value, content_type) = match &chain.source {
-                ChainSource::Request { recipe, trigger } => {
+                ChainSource::Request {
+                    recipe,
+                    trigger,
+                    component,
+                } => {
                     let response =
                         self.get_response(context, recipe, *trigger).await?;
                     // Guess content type based on HTTP header
                     let content_type =
                         ContentType::from_response(&response).ok();
-                    (response.body.into_bytes(), content_type)
+                    let value = match component {
+                        ChainRequestComponent::Body => {
+                            response.body.into_bytes()
+                        }
+                        ChainRequestComponent::Header(target_header) => {
+                            if let Some(val) =
```

Comment:
There's an important behavior decision to be made here, which is: What do we do if the header isn't in the response? Personally I think I'd rather have it error out. If you're looking for a header and it's not there, I think it's most likely that someone weird is happening and you want to know about it rather than have it just use an empty value. Maybe there's a use case for a header only appearing sometimes, and you just want to ignore the error. What do you think?

Reply:
Interesting question. My thought process was that instead of erroring out from the chain, naturally the request that requires the chained value is going to fail if the token is an empty value (at least one would hope). I'd be happy to implement it either way, I just liked the idea of the post-login request failing due to a bad token and then maybe the user can check their response from the chain request?

Reply:
Yeah that's true, in general I don't think the empty value would be too hard to debug, but maybe a bit harder than an up-front error. If we make it an error now though, and people find that annoying, it's a non-breaking change to replace the error with an empty value. But going the other way (default -> error) is a breaking change. So I'd say let's start with the error and see if we get any complaints.